src/main/kotlin/com/cjbooms/fabrikt/generators/client/ClientGeneratorUtils.kt

```diff
@@ -16,18 +18,24 @@ object ClientGeneratorUtils {
     const val ADDITIONAL_HEADERS_PARAMETER_NAME = "additionalHeaders"
 
     /**
-     * It resolves the API operation response to its body type. It iterates over all non-default responses
-     * by filtering those ones with content media. If multiple medias are found, then it will resolve to the schema
-     * reference of the first media type found, otherwise it'll resolve to Unit by assuming no response body
-     * for the given operation.
+     * Gives the Kotlin return type for an API call based on the Content-Types specified in the Operation.
+     * If multiple media types are found, but their response schema is the same, then the first media type is used and kotlin model for the schema is returned.
+     * If there are several possible response schemas, then the return type is JsonNode, so they are all covered.
+     * If no response body is found, Unit is returned.
      */
     fun Operation.toClientReturnType(packages: Packages): TypeName {
-        val returnType = this.getPrimaryContentMediaType()?.let {
-            toModelType(
-                packages.base,
-                KotlinTypeInfo.from(it.value.schema)
-            )
-        } ?: Unit::class.asTypeName()
+        val returnType =
+                if (hasMultipleResponseSchemas()) {
```

Comment:
Does this include the default Problem json, or is it just types outside of that?

Reply:
Ah it does, good catch. Fixed